src/yaml-language-support/yaml-support.ts

```diff
@@ -0,0 +1,68 @@
+import * as vscode from 'vscode';
+
+import { parse, findNodeAtPosition, util as yamlUtil } from 'node-yaml-parser';
+
+export interface YamlCache {
+    // the document model list represents the yaml text
+    yamlDocs: any[];
+
+    // lineLens contains the converted line length of each lines, it is used for converting from
+    // vscode position to the inner position in yaml element model.
+    lineLens: number[];
+
+    // the version of the document to avoid duplicate work on the same text
+    version: number;
+}
+
+export interface YamlMatchedElement {
+    // the document model list represents the yaml text
+    readonly yamlDocs: any[];
+
+    // the found node at the given position(usually at the edit/hover place)
+    readonly matchedNode: any;
+
+    // the document which contains the node at given position
+    readonly matchedDocument: any;
+}
+
+/**
+ * A yaml interpreter parse the yaml text and find the matched ast node from vscode location.
+ */
+ class YamlLocator {
+    
+    private _cache:{ [key:string]: YamlCache; }  = {};
```

Comment:
Perhaps document that this is mapping URIs to cached documents?

Reply:
yes